Sources/DSN.swift

```diff
@@ -8,88 +8,80 @@
 		self.projectID = projectID
 	}
-	
-	/// Can create a DSN struct from a string. If the string is not a
-	/// valid DSN format, a nil will be returned
-	public convenience init?(_ dsnString: String) {
+
+	/// Creates DSN object from a valid DSN string
+	internal convenience init?(_ dsnString: String) {
 		var dsn: NSURL?
 		var serverURL: NSURL?
 		var publicKey: String?
 		var secretKey: String?
 		var projectID: String?
-		
+
 		if let url = NSURL(string: dsnString),
-			host = url.host,
-			id = DSN.getProjectID(url) {
-				
-				// Setting properties
-				dsn = url
-				publicKey = url.user
-				secretKey = url.password
-				projectID = id
-				
-				// Creating components for serverURL
-				let path = "/api/\(id)/store/"
-				
-				// Setting componts to create NSURL
-				let components = NSURLComponents()
-				components.scheme = url.scheme
-				components.host = host
-				components.path = path
-				components.port = url.port
-				
-				serverURL = components.URL
+		host = url.host,
+		id = DSN.projectID(from: url) {
+
+			// Setting properties
+			dsn = url
+			publicKey = url.user
+			secretKey = url.password
+			projectID = id
+
+			// Setting componts to create NSURL
+			let components = NSURLComponents()
+			components.scheme = url.scheme
+			components.host = host
+			components.path = "/api/\(id)/store/"
+			components.port = url.port
+
+			serverURL = components.URL
 		}
 
 		guard let theDsn = dsn, theServerURL = serverURL, theProjectID = projectID else {
 			return nil
 		}
 		self.init(dsn: theDsn, serverURL: theServerURL, publicKey: publicKey, secretKey: secretKey, projectID: theProjectID)
 	}
-	
-	public typealias XSentryAuthHeader = (key: String, value: String)
-	
-	/// Creates a tuple with the header name and header value
-	public var xSentryAuthHeader: XSentryAuthHeader {
-		// Header parts
+
+	/// Tuple with the header name and header value
+	internal var xSentryAuthHeader: XSentryAuthHeader {
+
+		// Create header parts
 		let headerParts: [(String, String?)] = [
-			("Sentry sentry_version", String(SentryInfo.sentryVersion)),
-			("sentry_client", "sentry-swift/\(SentryInfo.version)"),
-			("sentry_timestamp", String(Int(NSDate().timeIntervalSince1970))),
-			("sentry_key", self.publicKey),
-			("sentry_secret", self.secretKey)
+				("Sentry sentry_version", String(SentryClient.Info.sentryVersion)),
+				("sentry_client", "sentry-swift/\(SentryClient.Info.version)"),
+				("sentry_timestamp", String(Int(NSDate().timeIntervalSince1970))),
+				("sentry_key", self.publicKey),
+				("sentry_secret", self.secretKey)
 		]
-		
-		// Combine parts into comma, delimited string
-		let value = headerParts.reduce([], combine: { (combined, keyValue) -> [String] in
-			guard let value = keyValue.1 else {
-				return combined
-			}
-			return combined + ["\(keyValue.0)=\(value)"]
-		}).joinWithSeparator(",")
-		
+
+		var ret: [String] = []
+		headerParts.filter() { $0.1 != nil }.forEach() { ret.append("\($0.0)=\($0.1!)") }
```

Comment:
I felt like using the `filter` was more verbose about the process, but I also prefer the `flatMap` in my day to day dev. I can change it if that's ok with you.